### PR TITLE
Fixes #24: Add application icons for mac and linux

### LINKED ISSUE #24: Change app title and icon
image

### PR DESCRIPTION
**Pull request by Foxinauci**
Adds icons for windows (macos/linux untested currently).

Icons changes based on the users default theme (can be changed later if the app gets it's own theme settings or something).

Icons also change dynamically (If user changes theme when app is open).

### CODE REVIEW

**Comment by Foxinauci** [comment]
Might of been a way to reopen the old one I'm not sure if so I apologize ;-;

**Comment by Foxinauci** [comment]
WAIT THERE'S TOO EXTRA EMPTY LINES HOW ;-;

**Review by SrIzan10 — APPROVED** [review]

**Review by hrfarmer — CHANGES_REQUESTED** [review]
icon on mac is WAY too big

image

**Comment by hrfarmer** [comment]
also maybe other people have different opinions but i'd just drop the dark mode icons imo, like no other app i can think of on desktop changes the icon like that, so it'd be pretty out of place

**Comment by brw** [comment]
Yea I think only having the light one would probably be better. I imagine the dark one is moreso for if the logo is ever used on a black/dark background.

**Comment by Foxinauci** [comment]
Alright sounds good so remove the dark icon and get smaller icon for macos? Only issue is I don't really have a way to test for macos.

**Comment by hrfarmer** [comment]
just remove the dark mode icons, ill fix the mac icon in a little bit

File: src/main/index.ts [hunk 1]
```diff
@@ -9,13 +8,24 @@ import { join } from "path";
 async function createWindow() {
   const [width, height] = getBounds();
 
+  const getIconPath = () => {
```
Comment (CaptSiro) [inline src/main/index.ts:11]: This ternary operator mayhem should be rewritten. Ternary operators are fine. They become hard to work with when they are nested. Prefer standalone functions. Consider rewriting as:

```typescript
function getIcon(): string {
  if (process.platform === 'win32') {
    return join(__dirname, '../../build/icon.ico');
  }

  if (process.platform === 'darwin') {
    return join(__dirname, '../../build/icon.icns');
  }

  return join(__dirname, '../../build/icon.png');
}

**Review by CaptSiro — CHANGES_REQUESTED** [review]

File: src/main/index.ts (hunk 1 above)
Comment (brw) [reply src/main/index.ts:11]: you can provide code as a suggestion btw if you select multiple lines and then press the first button in the toolbar or manually write
\`\`\`suggestion
\`\`\`

Comment (Foxinauci) [reply src/main/index.ts]: My apologies on that I thought ternary operators looked cleaner lol. Should be changed to what you sent.

**Review by CaptSiro — APPROVED** [review]

**Comment by CaptSiro** [comment]
The function should have been a global function. That's a minor detail and it really does not need addressing. Merging...

**Comment by Foxinauci** [comment]
Should be good on the conflicts end of things, only thing left I believe was for @hrfarmer to fix the macos icon being a little too big since I don't have a macos device to test on.

**Comment by hrfarmer** [comment]
i already fixed the macos icon, i just need someone else to test the pr for other platforms because git for the life of me won't let me check out this pr for some reason

**Comment by Foxinauci** [comment]
Ah I see, on Windows it worked fine for me.

**Comment by brw** [comment]
On Linux it only seems to work if I integrate the AppImage using something like AppImageLauncher, but that might just be an inherent limitation of AppImages (and even then it still doesn't show up in my dock for some reason, only on the executable/program itself). Not sure about the .deb or snap as I don't use either of those on Arch.

It might also be better to have a 1024x1024 version instead of a 256x256 version. Electron-builder should generate the other ones from it. Although you can also export all sizes directly from Figma and place them in the `build/icons` directory, naming them according to https://www.electron.build/icons.html, which should technically be better quality than downsampling one larger image.

**Review by CaptSiro — APPROVED** [review]

**Review by hrfarmer — APPROVED** [review]

**Pull request merged**